### PR TITLE
Replace sprintf with snprintf.

### PR DESCRIPTION
**Pull request by Hadatko**
Signed-off-by: Cervenka Dusan <cervenka@acrios.com>

# Pull request

**Choose Correct**

- [] bug
- [X] feature

**Describe the pull request**

Replacing sprintf with snprintf -> safety

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- eRPC Version:

**Steps you didn't forgot to do**

- [ ] I checked if other PR isn't solving this issue.
- [ ] I read Contribution details and did appropriate actions.
- [ ] PR code is tested.
- [ ] PR code is formatted.
- [ ] Allow edits from maintainers pull request option is set (recommended).

**Additional context**

### CODE REVIEW

**Comment by Hadatko** [comment]
This PR partially solve issue here: https://github.com/EmbeddedRPC/erpc/issues/342
Also change in erpc_sysgpio are not tested, and it these files are taken from somewhere it could be fine to contact owner and apply fix.

**Comment by Hadatko** [comment]
This PR partially solve issue here: https://github.com/EmbeddedRPC/erpc/issues/342
Also change in erpc_sysgpio are not tested, and it these files are taken from somewhere it could be fine to contact owner and apply fix.

File: erpc_c/port/erpc_sysgpio.c [hunk 1]
```diff
@@ -44,8 +46,8 @@ int gpio_export(int gpio)
         }
         else
         {
-            ret = sprintf(sysgpio, "%d", gpio);
-            if (ret >= 0)
+            ret = snprintf(sysgpio, SYSGPIO_LEN, "%d", gpio);
+            if ((ret >= 0) && (ret < SYSGPIO_LEN))
             {
                 if (strlen(sysgpio) != write(fd, sysgpio, strlen(sysgpio)))
```
Comment (amgross) [inline erpc_c/port/erpc_sysgpio.c:52]: I would replace both strlen(sysgpio) with ret

Comment (Hadatko) [reply erpc_c/port/erpc_sysgpio.c:52]: Very nice catch. Thank you ;)

**Review by amgross — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
> This PR partially solve issue here: #342 Also change in erpc_sysgpio are not tested, and it these files are taken from somewhere it could be fine to contact owner and apply fix.

OK, will do that

**Comment by MichalPrincNXP** [comment]
> > This PR partially solve issue here: #342 Also change in erpc_sysgpio are not tested, and it these files are taken from somewhere it could be fine to contact owner and apply fix.
> 
> OK, will do that

erpc_sysgpio author is OK with proposed changes

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by Hadatko** [comment]
great

**Comment by MichalPrincNXP** [comment]
thank you

**Pull request merged**